src/lib.rs

```diff
@@ -316,6 +356,38 @@ impl Member {
         quote!()
     }
 
+    /// Returns (format_string, argument) tuple.
+    fn defmt(&self) -> Option<(String, TokenStream)> {
+        let inner = self.inner.as_ref()?;
+
+        if inner.from.is_empty() {
+            return None;
+        }
+
+        // default to using {:?}
+        let mut spec = "{:?}".to_owned();
+
+        // primitives supported by defmt
+        const PRIMITIVES: &[&str] = &[
+            "bool", "usize", "isize", "u8", "u16", "u32", "u64", "u128", "i8", "i16", "i32", "i64",
+            "i128", "f32", "f64",
+        ];
+
+        // get the type name so we can use more efficient defmt formats
+        // if it's a primitive
+        if let syn::Type::Path(syn::TypePath { ref path, .. }) = inner.ty {
+            if let Some(ident) = path.get_ident() {
```

Comment:
I usually don't use the `ref` keyword anymore and just put an `&` before `inner.ty`.

```suggestion
        if let syn::Type::Path(syn::TypePath { path, .. }) = &inner.ty {
```